repo/manifest/committed_manifest_manager.go

```diff
@@ -98,31 +107,75 @@ func (m *committedManifestManager) commitEntries(ctx context.Context, entries ma
 // the lock via commitEntries()) and to compact existing committed entries during compaction
 // where the lock is already being held.
 // +checklocks:m.cmmu
-func (m *committedManifestManager) writeEntriesLocked(ctx context.Context, entries map[ID]*manifestEntry) (map[content.ID]bool, error) {
+func (m *committedManifestManager) writeEntriesLocked(ctx context.Context, entries map[ID]*inMemManifestEntry) (map[content.ID]bool, error) {
 	if len(entries) == 0 {
 		return nil, nil
 	}
 
-	man := manifest{}
+	switch m.formatVersion {
```

Comment:
should there be some heuristics where for trivially small number of entries we always write them in V0 format and only compaction uses V1?

Reply:
ah, I think I see why this would be a good idea. It seems like it would help reduce churn on the set of content pieces that the index is tracking. I've put in a quick update to do that, though like other things in the PR, the metrics I picked are sort of out of a hat. I'd be happy to take recommendations from folks if they have opinions